integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/mocks/oslpdevice/DeviceState.java

```diff
@@ -0,0 +1,105 @@
+    private final ConcurrentMap<MessageType, ConcurrentLinkedQueue<Oslp.Message>> receivedRequestsMap = new ConcurrentHashMap<>();
+
+    public DeviceState(final String deviceUID) {
+        this.deviceUID = deviceUID;
+    }
+
+    public Oslp.Message getMockedResponse(final MessageType messageType) throws DeviceSimulatorException {
+        return this.getFromMap(this.mockedResponsesMap, messageType);
+    }
+
+    public Oslp.Message getReceivedRequest(final MessageType messageType) throws DeviceSimulatorException {
+        return this.getFromMap(this.receivedRequestsMap, messageType);
+    }
+
+    public void addMockedResponse(final MessageType messageType, final Oslp.Message message) {
+        this.getQueue(this.mockedResponsesMap, messageType).add(message);
+    }
+
+    public void addReceivedRequest(final MessageType messageType, final Oslp.Message message) {
+        this.getQueue(this.receivedRequestsMap, messageType).add(message);
+    }
+
+    public boolean hasMockedResponses(final MessageType messageType) {
+        return !this.getQueue(this.mockedResponsesMap, messageType).isEmpty();
+    }
+
+    public boolean hasReceivedRequests(final MessageType messageType) {
+        return !this.getQueue(this.receivedRequestsMap, messageType).isEmpty();
+    }
+
+    public void incrementSequenceNumber() {
+        int numberToAddToSequenceNumberValue = 1;
+
+        if (ScenarioContext.current().get(PlatformKeys.NUMBER_TO_ADD_TO_SEQUENCE_NUMBER) != null) {
+            final String numberToAddAsNextSequenceNumber = ScenarioContext.current()
+                    .get(PlatformKeys.NUMBER_TO_ADD_TO_SEQUENCE_NUMBER)
+                    .toString();
+            if (!numberToAddAsNextSequenceNumber.isEmpty()) {
+                numberToAddToSequenceNumberValue = Integer.parseInt(numberToAddAsNextSequenceNumber);
+            }
+        }
+        int next = this.sequenceNumber + numberToAddToSequenceNumberValue;
+        if (next > SEQUENCE_NUMBER_MAXIMUM) {
+            final int sequenceNumberMaximumCross = next - SEQUENCE_NUMBER_MAXIMUM;
+            if (sequenceNumberMaximumCross >= 1) {
+                next = sequenceNumberMaximumCross - 1;
+            }
+        } else if (next < 0) {
+            final int sequenceNumberMaximumCross = next * -1;
+            if (sequenceNumberMaximumCross >= 1) {
+                next = SEQUENCE_NUMBER_MAXIMUM - sequenceNumberMaximumCross + 1;
+            }
+        }
+
+        this.sequenceNumber = next;
+    }
+
+    private ConcurrentLinkedQueue<Oslp.Message> getQueue(final Map<MessageType, ConcurrentLinkedQueue<Oslp.Message>> messageMap, final MessageType messageType) {
+        messageMap.computeIfAbsent(messageType, k -> new ConcurrentLinkedQueue<>());
+        return messageMap.get(messageType);
+    }
+
+    private Oslp.Message getFromMap(final Map<MessageType, ConcurrentLinkedQueue<Oslp.Message>> messageMap, final MessageType messageType)
```

Comment:
Hmm, indeed, the name is unclear. I changed it to pollMessage. Since that's what is does. It polls the queue (picks up first item and removes it from the queue)